lib/queue.js

```diff
@@ -1,33 +1,89 @@
 class Queue {
-    constructor() {
-        // this.store = ...
-        throw new Error("This method has not been implemented!");
+    constructor(size = 20) {
+        this.store = new Array(size);
+        this.front = this.back = -1;
+        this.size = size;
     }
 
     enqueue(element) {
-        throw new Error("This method has not been implemented!");
+        if ((this.back + 1) % this.size === this.front) {
+            throw newError("Full Queue!");
+        } else if (this.front === -1) {
+            //if empty 
+            this.front = 0;
+            this.back = 0;
+            this.store[this.back] = element;
+        } else if (this.back === this.size - 1 && this.front !== 0) {
+            this.back = 0;
+            this.store[this.back] = element;
+        } else {
+            console.log("HERE", this.store)
+            this.back = (this.back + 1) % this.size;
+            this.store[this.back] = element;
+        }; 
     }
 
     dequeue() {
-        throw new Error("This method has not been implemented!");
+        // let temp = null;
+
+        // let data = this.store[this.front];
+        // this.store[this.front] = null;
+
+        // if (this.front === -1) {
+        //     throw newError("Empty Queue!");
+        // } else if (this.front === this.back) {
+        // //    let temp = this.store[this.front];
+        //     this.front = -1;
+        //     this.back = -1;
+        //     // return temp;
+        // } else {
+        //     // temp = this.store[this.front];
+        //     this.store[this.front] = null;
+        //     this.front = (this.front + 1) % this.size;
+        //     // return temp;
+        // }
+        // return data;
+        if (this.front === -1) throw newError("Empty Queue!");
+
+        let data = this.store[this.front];
+        this.store[this.front] = null;
+
+        //if the queue is empty
+        if (this.front === this.back) {
+            console.log("DEQUEUE")
+
+            this.front = -1;
+            this.back = -1;
+        } else if (this.front === this.size - 1) {
+            console.log("DEQUEUE")
+
+            this.front = 0;
+        } else {
+            console.log("DEQUEUE")
+            this.front = (this.front + 1) % this.size;
+        }
+
+        return data;
     }
 
     front() {
-        throw new Error("This method has not been implemented!");
+        return this.store[this.front];
     }
 
     size() {
-        throw new Error("This method has not been implemented!");
+        return this.store.size;
     }
 
     isEmpty() {
-        throw new Error("This method has not been implemented!");
+       return this.front === -1;
     }
 
     toString() {
         let arr;
-        if (this.head > this.tail) {
-            arr = this.store.slice(this.head, this.capacity).concat(this.store.slice(0, this.tail));
+        if (this.front > this.back) {
+            arr = this.store.slice(this.front, this.size).concat(this.store.slice(0, this.back));
```

Comment:
Because of how you are implementing this. 

```suggestion
            arr = this.store.slice(this.front, this.size).concat(this.store.slice(0, this.back + 1));
```